aiida/cmdline/utils/migration/v03_to_v04.py

```diff
@@ -0,0 +1,528 @@
+    """
+    Migration of export files from v0.3 to v0.4
+
+    Note concerning migration 0032 - REV. 1.0.32:
+    Remove legacy workflow tables: DbWorkflow, DbWorkflowData, DbWorkflowStep
+    These were (according to Antimo Marrazzo) never exported.
+    """
+    old_version = '0.3'
+    new_version = '0.4'
+
+    verify_metadata_version(metadata, old_version)
+    update_metadata(metadata, new_version)
+
+    # Apply migrations on Nodes
+    if 'Node' in data['export_data']:
+        migration_base_data_plugin_type_string(data)
+        migration_process_type(metadata, data)
+        migration_code_sub_class_of_data(data)
+        migration_add_node_uuid_unique_constraint(data)
+        migration_migrate_builtin_calculations(data)
+        migration_provenance_redesign(data)
+        migration_dbgroup_name_to_label_type_to_type_string(metadata, data)
+        migration_calc_job_option_attribute_keys(data)
+        migration_move_data_within_node_module(data)
+        migration_trajectory_symbols_to_attribute(data, folder)
+        migration_remove_node_prefix(data)
+        migration_rename_parameter_data_to_dict(data)
+        migration_dbnode_type_to_dbnode_node_type(metadata, data)
+    else:
+        # Make needed changes to metadata.json (and check UUIDs - migration_14_and_18)
+        migration_process_type(metadata, data)
+        migration_add_node_uuid_unique_constraint(data)
+        migration_dbgroup_name_to_label_type_to_type_string(metadata, data)
+        migration_dbnode_type_to_dbnode_node_type(metadata, data)
+
+    # Apply migration on Groups
+    if 'Group' in data['export_data']:
+        migration_dbgroup_type_string_change_content(data)
+
+    # Apply common migrations
+    migration_remove_dbcomputer_enabled(metadata, data)
+    migration_replace_text_field_with_json_field(data)
+
+    # Update metadata.json with the new Log and Comment entities
+    new_entities = {
+        "Log": {
+            "uuid": {},
+            "time": {
+                "convert_type": "date"
+            },
+            "loggername": {},
+            "levelname": {},
+            "message": {},
+            "metadata": {},
+            "dbnode": {
+                "related_name": "dblogs",
+                "requires": "Node"
+            }
+        },
+        "Comment": {
+            "uuid": {},
+            "ctime": {
+                "convert_type": "date"
+            },
+            "mtime": {
+                "convert_type": "date"
+            },
+            "content": {},
+            "dbnode": {
+                "related_name": "dbcomments",
+                "requires": "Node"
+            },
+            "user": {
+                "related_name": "dbcomments",
+                "requires": "User"
+            }
+        }
+    }
+    metadata['all_fields_info'].update(new_entities)
+    metadata['unique_identifiers'].update({"Log": "uuid", "Comment": "uuid"})
+
+    # Update data.json with the new Extras
```

Comment:
Probably put this in a separate function as well, even though it does not correspond to an actual database migration

Reply:
For the `Log` and `Comment` entries as well? (The lines above this comment)

Reply:
Only moved the introduction of the extras dicts for now